### PR TITLE
Add a single variables slicing for subcell

### PR DESCRIPTION
**Pull request by yoonso0-0**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

I'm not sure `std::copy`ing from `std::vector<double>` into the returned variables object is the best option for implementing this feature. There might be more efficient way, so welcome any suggestions.

### CODE REVIEW

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#include "Utilities/Gsl.hpp"
+
+namespace evolution::dg::subcell {
+/// @{
+/*!
+ * \brief Slice a Variables object on subcell mesh for a given direction and
+ * slicing depth (number of ghost points).
+ *
+ * This is essentially a wrapper of
+ * `evolution::dg::subcell::detail::slice_data_impl` function, but note that the
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp]: I don't know how great it is to refer to a `detail` function in the documentation. It's a detail for a reason, not a public interface. Isn't the contrast to the other `slice_data`?

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * last argument has the type `Direction<Dim>`, not a DirectionMap.
+ */
+template <size_t Dim, typename TagList>
+void slice_variable_for_subcell(
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp]: why add `for_subcell`, isn't that redundant with being in the `subcell` namespace? It's not clear what you are trying to differentiate here

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const gsl::not_null<Variables<TagList>*>& sliced_subcell_vars,
+    const Variables<TagList>& volume_subcell_vars,
+    const Index<Dim>& subcell_extents, const size_t ghost_zone_size,
+    const Direction<Dim>& direction) {
+  // check the size of sliced_subcell_vars (output)
+  const size_t num_sliced_pts =
+      subcell_extents.slice_away(direction.dimension()).product() *
+      ghost_zone_size;
+  if (sliced_subcell_vars->size() != num_sliced_pts) {
+    *sliced_subcell_vars = Variables<TagList>(num_sliced_pts);
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp]: `sliced_subcell_vars->initialize(num_sliced_pts);`

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  // Slice volume variables. Note that the return type of slicing function is
+  // std::vector<double>.
+  DirectionMap<Dim, bool> directions_to_slice{};
+  directions_to_slice[direction] = true;
+
+  std::vector<double> sliced_data{detail::slice_data_impl(
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp]: `const`?

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+template <size_t Dim, typename TagList>
+Variables<TagList> slice_variable(const Variables<TagList>& volume_subcell_vars,
+                                  const Index<Dim>& subcell_extents,
+                                  const size_t ghost_zone_size,
+                                  const Direction<Dim>& direction) {
+  Variables<TagList> sliced_subcell_vars{};
+  sliced_subcell_vars.initialize(
+      subcell_extents.slice_away(direction.dimension()).product() *
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp:57]: just pass the size into the variables constructor

File: src/Evolution/DgSubcell/SliceVariable.hpp [hunk 6]
```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/SliceVariable.hpp:7]: `<vector>`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/Evolution/DgSubcell/SliceVariable.hpp (hunk 1 above)
Comment (yoonso0-0) [reply src/Evolution/DgSubcell/SliceVariable.hpp]: Agree. I'll delete this paragraph.

**Comment by yoonso0-0** [comment]
Pushed a fixup. Thanks for the review :)

**Review by nilsdeppe — COMMENTED** [review]
LGTM, please rebase, squash, and ping me when you've done that :)

**Comment by yoonso0-0** [comment]
@nilsdeppe Rebased and squashed :)

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**